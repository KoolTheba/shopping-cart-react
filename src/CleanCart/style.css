.clean-cart-button {
    width: auto;
    height: 40px;
    margin-top: 2rem;
    font-size: 1rem;
    font-weight: bold;
    padding: 10px;
    cursor: pointer;
}

.x-icon {
    margin-left: 0.5rem;
}

.clean-cart-button:hover {
    background-color: #dd3e54;
}