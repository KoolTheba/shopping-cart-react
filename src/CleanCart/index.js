import React from 'react'
import './style.css'

function CleanCart () {
    return(
        <>
        <button>Clean Cart</button>
        </>
    )
}

export default CleanCart