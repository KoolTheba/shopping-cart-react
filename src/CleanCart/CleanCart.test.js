import React from 'react'
import CleanCart from './'
import { shallow } from 'enzyme'

const props = {
    cleanShoppingList: jest.fn()
}

describe('CleanCart component', () => {
    it('renders CleanCart component correctly', () => {
        const wrapper = shallow(<CleanCart {...props}/>)
        expect(wrapper.find('.clean-cart-button')).toHaveLength(1)
        expect(wrapper.find('.x-icon')).toHaveLength(1)
        expect(wrapper).toMatchSnapshot()
    });

    it('when clicking clean cart button, cleanShoppingList() is called', () => {
        const wrapper = shallow(<CleanCart {...props}/>)
        const cleanListButton = wrapper.find('.clean-cart-button')
        cleanListButton.simulate('click')
        expect(props.cleanShoppingList).toHaveBeenCalledTimes(1)
    });
})
