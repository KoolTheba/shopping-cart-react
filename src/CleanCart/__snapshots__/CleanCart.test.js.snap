// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CleanCart component renders CleanCart component correctly 1`] = `ShallowWrapper {}`;
